src/inmanta/module.py

```diff
@@ -1988,10 +1990,13 @@ def verify_python_requires(self) -> None:
         """
         Verifies no incompatibilities exist within the Python environment with respect to installed module v2 requirements.
         """
-        if not env.ActiveEnv.check(in_scope=re.compile(f"{ModuleV2.PKG_NAME_PREFIX}.*")):
-            raise CompilerException(
-                "Not all installed modules are compatible: requirements conflicts were found. Please resolve any conflicts"
-                " before attempting another compile. Run `pip check` to check for any incompatibilities."
+        constraints: List[Requirement] = [Requirement.parse(item) for item in self.collect_python_requirements()]
+        constraints_for_marker = filter(lambda c: not (c.marker and not c.marker.evaluate()), constraints)
```

Comment:
Wouldn't it be better to check for markers in `ActiveEnv.check` where `constraint_violations` is built?

Reply:
Why would it be better?

Reply:
Because this is actually a bug in the `check` implementation rather than something specific to this verification.